### ISSUE TITLE
Unreliable MagiskHide Process Monitor

### ISSUE THREAD

**Issue by topjohnwu** [topjohnwu/Magisk#907]
For some people, the MagiskHide process monitor is not 100% reliable, which leads to random root detections on hidden apps.

**Comment by lialosiu** [comment]
Until now, All the devices have this issue is Oneplus...

Maybe some special things in Oneplus H2OS/OOS system?

**Comment by paarnoutse** [comment]
An XDA user reported at https://forum.xda-developers.com/showpost.php?p=78498587&postcount=29972 that disabling "phone" permission for Google Play made it working for him/her.

I didn't check this myself (I currently reverted to a previous Magisk version to prevent root detections), but possibly a useful tip for finding the reason for the issue in Magisk v18.

**Comment by hkalltheway** [comment]
Not only oneplus it seems. I am having this issue with the Galaxy Note 9. Also, I don't know if it's related, but everytime I open outlook, safetynet check fails until I restart the device

**Comment by Madis0** [comment]
Fails on 9.0 OmniROM (OnePlus 5T) - Response payload validation failed. Play Store is certified.
Worked on previous OmniROM and Magisk.

**Comment by deception83** [comment]
Fails on my Stock ROM Pixel XL too.

**Comment by DKW-LK** [comment]
My problem was disabling magskhide after minutes enabling it... It was 17.1 What I did was i removed some apps and kept 3 to 4 apps only in magiskhide list... It worked... After flashing uninstaller zip and then again installing 18.0 it works for me grate now.. Im on miui 10 Whyred.... (8.1.0 oreo)

**Comment by null4n** [comment]
> An XDA user reported at https://forum.xda-developers.com/showpost.php?p=78498587&postcount=29972 that disabling "phone" permission for Google Play made it working for him/her.
> 
> I didn't check this myself (I currently reverted to a previous Magisk version to prevent root detections), but possibly a useful tip for finding the reason for the issue in Magisk v18.

Thanks a lot, this works for me. After clear play store user data and disable it's phone permission, all the safetynet detect app now can be shown as before (version 17.1)

**Comment by gypsophlia** [comment]
this problem occur on my phone after Magisk was upgraded to 18.0.
After using the phone for a while, it can not pass SaftNet. However, After cleaning back ground process, MagiskHide works again. 

MagiskHide works find with Magisk Version17.1

ROM MIUI GLOBAL 10.0 
Phone MI 6
Magisk version 18.0

**Comment by sfai05** [comment]
> Fail on Pixel 3 after running for a while

Same here. Reverted back to Magisk 17.3 and it works fine

**Comment by DocJaat** [comment]
I am using Oneplus 5 running Oxygen OS Pie. I had no problems with Magisk 17.1 on Oreo but since dirty flashing to Pie and also updating Magisk to Version 18 - I am finding that Safetynet passed initially - after phone is in standby for some time - safetynet fails (both ctsProfile & basicIntegerity) - after reboot safetynet passes again. Also if I go to Magisk Manager and turn off Magisk Hide and switch it back on again - then the Safetynet paases again - only for sometime - after standby it fails again. Any solution?

**Comment by hamid-elaosta** [comment]
On a OnePlus 6T. Doesn't work with 18.0, works fine after flashing back to 17.3.

On 18.0 it appears to be intermittent; after a reboot, SafetyNet will pass for a little while, then it will not until another reboot. I'll stick with 17.3 for now.

**Comment by xksefa** [comment]
I've the same issue with Galaxy S7Edge. Android 8.0.0. Stock ROM. I need to disable MagiskHide and Re-Enable it to make it worked properly.

**Comment by skaforey** [comment]
I can confirm that disabling and enabling Magisk Hide fixes this.  Is there a way to do this via the command line? I would like to setup a tasker profile to do this.  On my Pixel 2 XL I am having to do this like 5 times per day as the daemon(?) seems to stop running frequently for me

**Comment by Didgeridoohan** [comment]
@skaforey Take a look in the docs: https://topjohnwu.github.io/Magisk/tools.html

**Comment by Danz17** [comment]
I report same issue, with one plus 6 pie, few apps detect root, even on hide, intune as well, installed intune hider module .. worked for solid 10 min thing again detected,

Version of magisk used is 18

**Comment by sbaum** [comment]
Same problem (both SafetyNet checks failing sometimes) on my Pixel 2 XL with latest Android and Magisk 18.0. Now trying if Magisk 17.3 works reliable...

**Comment by dalm3000** [comment]
I have this problem too, passes after toggle. OP6T but funny thing is, when failing safetynet everything that checks for root/safetynet still works. I don't even bother toggling anymore. Google pay, PoGo, bank apps all working with safetynet failing. Odd. It does pass on boot though.

**Comment by JRoy** [comment]
Device: Pixel 1
Android Version: 9
Security Patch: January 5th, 2019
Magisk: v18.1-1df65940

SafetyNet passes for about 10-20 minutes after a reboot, after which it fails. Toggling MagiskHide fixes this issue temporarily

**Comment by AlphaDE** [comment]
On OP 6, OOS 9.0.3, deleting cache/memory and removing phone permission of the Play Store doesn't help.

**Comment by AlphaDE** [comment]
I downgraded to Magisk 17.2 and Safetynet status check is successful again and MagiskHide works.

**Comment by sapientsaxonsaboo** [comment]
This was originally raised [here](https://github.com/topjohnwu/Magisk/issues/859#issue-390155234) and a thread on XDA.

--SSS

**Comment by shellster** [comment]
I'm also experiencing the problem on a Nexus 6P and a Pixel 2

**Comment by robross0606** [comment]
Same problem on Galaxy S8

**Comment by asumos** [comment]
Running into this issue on Pixel 3 XL Build PQ1A.181205.006

**Comment by fredskis** [comment]
Been having this issue on my OnePlus 6. Happens on both stable OOS 9.0.3 and the latest beta 10.

**Comment by alpetrus** [comment]
Moto G6 Plus (evert), LOS 16 + Havoc OS. 
ctsProfile and basic Integrity fails after some times of usage phone. Reboot solved the problem. 
Downgrade do Magisk 17.3 solve the problem permanently.

**Comment by jamesli1258** [comment]
Also occurs on my Pixel 2

**Comment by fawkes15** [comment]
Confimerd on OnePlus 5T (Android 9 OSS beta 23)

**Comment by myself379** [comment]
I'm on Xiaomi A1 Oreo 8.1 facing smiliar issue. I even tested with Magisk core only on Magisk v18

**Comment by FernanDK** [comment]
Same problem here, Motorola Z2 stock ROM, I have to reboot system to pass safety net, and sometimes toggling magisk hide doesn't work so I have to reboot again.

I've noticed this behaviour is prominent to happen after automatic updates from play store kick in whilst charging the phone. Meaning play store could be the real vilan here.

**Comment by lss4** [comment]
I'm on XZ Premium with CarbonROM and having similar issues.
The phone currently cannot pass ctsProfile (only basicIntegrity), but after a while (usually when the phone went to sleep for an extended amount of time), basicIntegrity won't pass anymore. And when that happens, games that detect root would refuse to start, indicating Magisk Hide has been broken.

Toggling Magisk Hide on and off could fix it for a while, but due to the amount of apps I added into the Magisk Hide, toggling it would result in restart of SystemUI as well as reinitialization of SIM card (and occasionally cellular data couldn't be brought up and I have to toggle it again).

I suspect services/threads Magisk Hide use might get killed when the phone goes to sleep for an extended amount of time so not sure if these could be made into something privileged that'll stay active, at the expense of battery life, to see if this makes a difference. If not, then the possibility that the background process management might be involved would be ruled out.

**Comment by Habylab** [comment]
Does the latest Canary build fix this? I've seen a couple suggest so.

**Comment by raz4725** [comment]
tried a few weeks ago on a pixel 3... same issue

**Comment by Pitel** [comment]
Same issue. Nexus 5X, LineageOS 15.1 (some fo the laatest builds).

`ctsProfile` is false, `basicIntegrity` is true. Disabling MagiskHide makes boths false.

Pokemon Go works, but can't pay with Google Pay.

**Comment by Habylab** [comment]
Confirming first and latest 18.1 Canary doesn't fix the issue. It's been acknowledged in stable release notes though.

**Comment by xksefa** [comment]
Confirming it is not solved on 18.1 but I couldn't see any statement saying it is solved in release notes.

**Comment by diemadedrei** [comment]
Same problem here on oneplus 6, lineageOS 16 but only since 18.1 it was working with 18. It seems to stop working the moment my payment app checks safetynet.

**Comment by amanteae** [comment]
anyone using magisk with adguard? I found adguard might break magisk hide, but thy fixed in the yesterday nightly build. 

![image](https://user-images.githubusercontent.com/21978129/52393051-6ad61780-2adf-11e9-9a5f-63ed0d434777.png)

**Comment by xksefa** [comment]
I was using with AdAway not Adguard. But still the same issue.

**Comment by James-Oakley** [comment]
Is that Adguard issue related to _this_ issue with Magisk?

**Comment by amanteae** [comment]
I still believe it's Magisk issue. i still have safetynet failure after upgrading to the adguard nightly build. I'm using magisk canary build btw.

**Comment by AlphaDE** [comment]
I find it strange that new versions are released and the author is not even reacting to this major issue, making Magisk 18.x useless fir me and many others..

**Comment by AlphaDE** [comment]
I find it strange that new versions are released and the author is not even reacting to this major issue, making Magisk 18.x useless fir me and many others..  

And it's more than the title states, on my device, MagiskHide isn't working at all.

**Comment by shellster** [comment]
An update.  I've been back on 17.3 for about a month now, and it has been rock solid.  Zero root detections.  As far as I'm concerned, there is definitely a bug that was introduced in v18+ releases.

**Comment by papjul** [comment]
Hi @topjohnwu 
Could you update us on your findings?
I was thinking that this might be due to a change in target API. I know that latest API have different battery saving options, which may result in a more aggressive background killing of process/services, such as MagiskHide.

**Comment by amanteae** [comment]
> An update. I've been back on 17.3 for about a month now, and it has been rock solid. Zero root detections. As far as I'm concerned, there is definitely a bug that was introduced in v18+ releases.

Magisk hide is ok on 17.3, but repacking the magisk manager is not ok. I haven't tried magisk manager v7 + magisk v17.3 yet.

**Comment by papjul** [comment]
> 
> 
> > An update. I've been back on 17.3 for about a month now, and it has been rock solid. Zero root detections. As far as I'm concerned, there is definitely a bug that was introduced in v18+ releases.
> 
> Magisk hide is ok on 17.3, but repacking the magisk manager is not ok. I haven't tried magisk manager v7 + magisk v17.3 yet.

You can't. Magisk Manager V7 dropped support for Magisk < 18.0: https://github.com/topjohnwu/Magisk/blob/v18.1/app/src/full/res/raw/changelog.md

**Comment by dalm3000** [comment]
You can repackage magisk manager on 17.3 if you use the manager app from 18.0, just pull it from the zip. If you restore original package it will wipe with the version that doesn't work, but you can just reinstall using the zip from 18.0 zip.

**Comment by diemadedrei** [comment]
Definitely has nothing to do with ad modules as I have never used any of them.

**Comment by jwinstonsf** [comment]
I did a few things all at once. One or more of them fixed the problem. 
1. I got rid of the Safetypatcher module
2. I unhid everything except for Google Pay, Google Pay Send, Google Play Store, Google Play Services Framework, and Market Feedback Agent. You have to show system apps to see all of these.  (I was under the mistaken impression that I need to hide everything Google but apparently not!)
3. I  cleared all data from the Play store.

Everything is green for the last twelve hours and Google Pay worked at my local Safeway. It took a few hours but The Play Store shows my device as certified too. I hope this helps.

**Comment by RafaelDeJongh** [comment]
Updated to the latest version of my Pocophone F1 after some weird stuff started happening after I updated to the latest version of Magisk 18.1, first it started saying almost every time I connected to the internet that there was a new version was available, when I went into the Magisk Manager it said it required further setup, when pressing install it just popped up with an error "setup failed".

I then wiped all cache, uninstalled Magisk completely, updated to the latest firmware and Lineage OS build for my phone and re-installed Magisk, now I noticed that most if not all my apps that were originally hidden (even though I've checked the apps in Magisk Hide PLUS enabled the option to hide Magisk (repack it)) made sure to remove all zips/folders or whatever that had Magisk in its name, made sure to remove the cache, but sadly nothing has worked.

Now my banking apps and verification apps won't work anymore as they detect root as well as certain games like PoGO and others.

Does anyone have any idea how to solve this or get around this?

**Comment by cousinit99** [comment]
Version 18 was working okay. I don't remember having this issue.  Then I upgraded to 18.1, and safetynet starts failing left and right.  So, I uninstall 18.1 and MM 7.0 completely and install 18.0 again.  Safetynet is still failing!  Is there some remnant that the uninstaller didn't catch?

What I don't understand is why my root-a-phobe apps aren't complaining, even though safetynet is failing...

**Comment by DeliriousMetro** [comment]
> I did a few things all at once. One or more of them fixed the problem.
> 
> 1. I got rid of the Safetypatcher module
> 2. I unhid everything except for Google Pay, Google Pay Send, Google Play Store, Google Play Services Framework, and Market Feedback Agent. You have to show system apps to see all of these.  (I was under the mistaken impression that I need to hide everything Google but apparently not!)
> 3. I  cleared all data from the Play store.
> 
> Everything is green for the last twelve hours and Google Pay worked at my local Safeway. It took a few hours but The Play Store shows my device as certified too. I hope this helps.

Done this and hid those Google applications mentioned and locked the process on Oxygen OS for my OP6 and on miui on Mi 6X.

No errors so far. 

![screenshot_20190210-202210](https://user-images.githubusercontent.com/9902130/52539092-16969600-2d72-11e9-99f4-7fd29be0341b.jpg)

**Comment by RafaelDeJongh** [comment]
Update on my site:

I've used the Magisk Uninstaller once again,
Flashed to the latest framework again
Dirty Flashed to the latest Lineage OS again
Installed the latest Magisk Build 
Repacked the Magisk Application
Use the Magisk Hide to hide pretty much everything
Made sure to remove all saved profile data and cache from the apps to then also uninstall/reinstall them to be extra sure.

At the current time and after various reboots it seems to be working with the apps I mentioned that weren't anymore. I am not sure if this is temporary or will be consistent but it's at least something!

**Comment by DeliriousMetro** [comment]
Spoke too soon. 

Still broke and temporarily working. Not too sure why banking apps don't care about this... 

![screenshot_20190211-185959](https://user-images.githubusercontent.com/9902130/52586403-57a0b000-2e2f-11e9-864c-a9e4f3d87140.jpg)

**Comment by amanteae** [comment]
Staying on v17.3 is still the best solution for now.

**Comment by AlphaDE** [comment]
+1

**Comment by piit79** [comment]
> ... locked the process on Oxygen OS for my OP6 and on miui on Mi 6X.

This seems to have "fixed" (or rather, worked around) the problem. Locking the Magisk Manager process in Recents ensures MagiskHide doesn't stop working every couple of hours. While both SafetyNet checks still start failing after a while none of my apps (Barclays, Revolut, Google Pay) care about that and work just fine.

**Comment by paarnoutse** [comment]
> 
> 
> > ... locked the process on Oxygen OS for my OP6 and on miui on Mi 6X.
> 
> This seems to have "fixed" (or rather, worked around) the problem for me on OnePlus 6T. Locking the Magisk Manager process in Recents ensures MagiskHide doesn't stop working every couple of hours. While both SafetyNet checks still fail after a while, none of my apps (Barclays, Revolut, Google Pay) actually care about that and work just fine.

To prevent that Magisk Manager could be killed, since a long time at my device Magisk Manager I had already set "Don't Optimize" in Battery settings for it and also set it as 'locked' in Recent apps. However, unfortunately it doesn't prevent the regular root detections for hided apps in Magisk v18 at my device (OnePlus 5 running OxygenOS 5.1.7 ).

**Comment by zunurain222** [comment]
for now i rather user version 17.3 as currently it has no issues at all

**Comment by badabing2005** [comment]
Magisk Hide is causing a different sort of problem for me.
In my case it's not Safety Net that is not working, but rather Magisk hide hiding Magisk from Magisk Manager.
PS: I haven't repackaged MM to hide it, I'm aware of [this](https://www.didgeridoohan.com/magisk/Magisk#hn_Updating_from_v18.0_to_v18.1) issue, which is not the case for me.

When I first reboot, Magisk Manager works fine and detects Magisk as installed,
At first Safety Net fails (if I check immediately after reboot)
however after a little while it passes (I'm assuming that there is a delayed start of Magisk hide, so safety net fails until Magisk Hide kicks in, just a guess)

However once Magisk Hide kicks in, it causes MM not to see Magisk as installed (root apps continue working), 

If I kill MM or wait a while (I suppose battery optimization is killing it), and start it, MM does not detect Magisk as installed and prompts me to install, but Safety Net still passes.

The only way I can get MM working again is
Reboot and start MM before Magisk Hide kicks in
Or
- Disable Magisk Hide ( /sbin/magiskhide --disable)
- Start MM (Make sure MM is started before enabling Magisk Hide again)
- Enable Magisk Hide (/sbin/magiskhide --enable)
- Go back to MM settings and enable Magisk Hide

As previously stated, in my case Magisk Hide is hiding magisk, saftey net passes when MH is enabled.
Its just doing too good of a job and hiding Magisk from MM.

Thanks
BB

**Comment by skaforey** [comment]
I finally caved and reverted back to 17.3.  It's hard to complain about a free product that has been worked on for so many hours, but it would be nice if there was an update on any progress being made or ideas for people to try?

**Comment by AlphaDE** [comment]
I learned in xda-developers to monitor the author's Twitter thread and here the commits.. 

Btw I found on Twitter that John set up a Patroneo page where one can support his work.

**Comment by topjohnwu** [comment]
Rejoice guys!
Commits e5940168fe5d61d4ba34a9fad07b22fc61682c25...6412bfc7b5cbf42e23e6aff9d9cf718061008803 address this issue with a 100% rewritten from scratch MagiskHide implementation. Huge kudos to @arter97 for the contribution!

**Comment by asumos** [comment]
Yay it's fixed. When will we see 7.0.1?

**Comment by Pitel** [comment]
Not fixed! It's broken now. See xda for details.

**Comment by James-Oakley** [comment]
@Pitel - Could you please link to the relevant post on XDA?

**Comment by JRoy** [comment]
Note to people updating: remove everything from magisk hide before updating!

**Comment by asumos** [comment]
how do we update? download a new 7.0 from release thread? i am not seeing an option in app to update to a new version

**Comment by JRoy** [comment]
@asumos you need to be on canary channel

**Comment by asumos** [comment]
@JRoy ahhh okay thx. guess i will make the switch manually later..... or just wait either or, thanks again.

**Comment by amanteae** [comment]
> how do we update? download a new 7.0 from release thread? i am not seeing an option in app to update to a new version

If you are on v18.x , just manually switch the update channel to Custom and input the Magisk Canary link.  If you are on v17.3 or below, you need to uninstall the magisk and reinstall v18.x first. Then upgrade to magisk canary. Because Magisk Manager v7 doesn't support v17.x,  manager will seize to work on Magisk V17. Therefore it won't let you to upgrae magisk.

**Comment by Pitel** [comment]
@James-Oakley https://forum.xda-developers.com/apps/magisk/dev-magisk-canary-channel-bleeding-edge-t3839337, look at the latest posts.

And I also has both safety net check false now, while having them pass before 18106.

**Comment by DeliriousMetro** [comment]
> @James-Oakley https://forum.xda-developers.com/apps/magisk/dev-magisk-canary-channel-bleeding-edge-t3839337, look at the latest posts.
> 
> And I also has both safety net check false now, while having them pass before 18106.

On my old Xperia Z5, we used a safetynet cleaning tools flashable zip which removed drm folders from the data partition.

Might do the job like a clean install?

https://drive.google.com/file/d/0BxKcqvQU0FNMZnV3ZjdfQjU3NXM/view

**Comment by Didgeridoohan** [comment]
@Pitel There's practically zero chance of this being caused by the same bug that this issue was about. If there's a new issue, it would be good if those affected could investigate and gather information on what that could be (you know, logs and stuff). Since MagiskHide has been completely remade it's not strange if there are new bugs popping up.

But, before anything else: have you tried turning MagiskHide off and on again? If MagiskHide fails after a Magisk update that has been known to fix it.

**Comment by sapientsaxonsaboo** [comment]
Been testing on canary channel since release and so far, so good. Magisk Hide process continues to function reliably and no need to toggle off and on. 

-- SSS

**Comment by Pitel** [comment]
I probably was different bug, but 18107 build works fine again. :+1:

**Comment by amanteae** [comment]
I'm currently using 18106 for almost 2 days. No safetynet failure for now.
-Device: Pixel3

**Comment by shellster** [comment]
Canary is not working for me.  Magisk keeps saying it isn't installed after a bit, and I can't use root.  Going to revert back to 17.3 yet again.

**Comment by mbze430** [comment]
I am on an ASUS ROG phone and on 18.1 (18100) and safetynet fails after a few hours but a reboot fixes it

**Comment by amanteae** [comment]
> I am on an ASUS ROG phone and on 18.1 (18100) and safetynet fails after a few hours but a reboot fixes it

Try the newest magisk canary build v18.2 (18109). The magisk hide issue is fixed.

**Comment by James-Oakley** [comment]
I've installed latest (18114) to see if this issue is fixed. I can't say whether or not I'm encountering the _same_ issue, but Safetynet failed straight away, and again after a reboot.

**Comment by Didgeridoohan** [comment]
@James-Oakley What happens if you toggle MagiskHide offc and on?

**Comment by James-Oakley** [comment]
@Didgeridoohan: I wasn't sure whether you meant toggling Hiding Magisk (renaming the package name), or the "Magisk Hide" toggle further down the settings screen. The former didn't help (and was a fiddle, because each time you turn that on / off it restores the update channel to Stable). But the latter has fixed it for now. Is that a known issue, where you sometimes have to toggle that off and on after installing? Does it recur, or only once after first install?

**Comment by Didgeridoohan** [comment]
@James-Oakley Yes, MagiskHide not working after an update can happen (especially when we're talking about such major changes as is currently being introduced). Toggling it off and on (not repackaging the Manager)  it and everything's dandy.

**Comment by Djtrip83** [comment]
Hi there! Last weekend out of the blue, the Tap & Pay on my Pixel 2 Jan 2019 update stopped working and detect root even though Magisk 18 was passing SafetyNet. Also, 2 root sensitive apps were not allowing Fingerprint login and I had to toggle MagiskHide each time. I had BusyBox so I thought I must have edited /system at some point...

Today I updated the Pixel 2 to Mar 2019 release via clean install (applied Feb update first) and then I installed Magisk 18.1. From the start I had the same issue with the 2 root sensitive app (cannot test Tap & Pay as I am at home), even though I had cleared data/cache for each AND SafetyNet says test passed. The 2 apps would detect root around 5 mins after toggling MagiskHide and opening the app successfully.

Comments:
- toggling USB debugging did not make a difference
- SafetyNet was showing green, then red and then green again (toggled MagiskHide each time without rebooting)
- removing any MAGISK suffix from filenames on the Internal Storage seemed to have kept the 2 apps ok in the last 30 mins (maybe unrelated)
- clearing from recent apps the sensitive app before opening another one as well seemed to have kept the 2 apps ok in the last 30 mins (maybe unrelated)
- rooting is fine and working as in I don't loose root

I will monitor now and feed back.

PS: if this is not the right place for such lengthy feedback, let me know (I am used to XDA forums)

**Comment by James-Oakley** [comment]
@Djtrip83 You say you installed 18.1. Do you mean 18100?
If so, then yes it has a bug (the one being discussed here) where Safetynet randomly keeps dropping out. The solution is to use the latest canary build that fixes  that issue.
Or are you already using 18114 or similar?

**Comment by Djtrip83** [comment]
I am using 18100.
18114 fixes the issue?

**Comment by James-Oakley** [comment]
> I am using 18100.
> 18114 fixes the issue?

That's the claim, yes: https://github.com/topjohnwu/Magisk/issues/907#issuecomment-464308411

I'm just testing that myself. For a long while I stayed on 17.x to avoid this problem which first appeared in the 18.x releases. 3 weeks ago there was a major commit that claims to fix the problem. Others are reporting that it has, so I've just now switched over to Canary to try this for myself.

**Comment by shellster** [comment]
I don't know if it is the same issue or not, as safety net still passed, but after the latest canary build update, Outlook is detecting root again.  It is checked under MagiskHide.  I had to revert back to 17.3 again.

**Comment by alescdb** [comment]
Same here, safetynet pass but can't make a payment, a message saying "you cannot pay in shop anymore with Google Pay" (translate from French). 
My last successful payment was on March 2nd (v17.3)

**Comment by Djtrip83** [comment]
> > I am using 18100.
> > 18114 fixes the issue?
> 
> That's the claim, yes: [#907 (comment)](https://github.com/topjohnwu/Magisk/issues/907#issuecomment-464308411)
> 
> I'm just testing that myself. For a long while I stayed on 17.x to avoid this problem which first appeared in the 18.x releases. 3 weeks ago there was a major commit that claims to fix the problem. Others are reporting that it has, so I've just now switched over to Canary to try this for myself.

@James-Oakley which Play Services version do you have? And is it stable or beta? I heard of an issue coming from Play Services v15 and above so I just downgraded to Play Services 14.7.99 and kept Magisk 18100. I had to re-setup Google Pay (confirm card, etc.) and no root warning came up (which I take as a good sign). I will see how it goes but wanted to share it with you.

**Comment by fiatCurrency** [comment]
Not sure how related this is. GP hasn't worked on my S5 for over a month. SafetyNet passes. Noticed that there is a _lot_ of logcat traffic (as in 1000+ lines  a minute).

**Comment by hongtaat** [comment]
Just a little sharing here..
After a clean flash to Mokee ROM (both Oreo and Pie) on my Mi 3W (cancro) and install Magisk, both ctsProfile & basicIntegerity are false. Both will only become true when I toggled MagiskHide, and will remain that way until the phone is rebooted. So every reboot I have to re-toggle MagiskHide.
What I did is add the following script in service.d
`#!/system/bin/sh
sleep 20
magiskhide --disable
sleep 2
magiskhide --enable`
And that somehow save me the trouble of manually re-toggling.
